### PR TITLE
docs: improve datadog integration docs

### PR DESCRIPTION
**Pull request by daveleek**
## About the changes

Improvement to the description of the datadog integration. Adds 2 missing event types, removes an event type that is deprecated and about to be completely removed, adds missing description of extra json headers and source type name, and adds description for the new configuration option for JSON body support

### CODE REVIEW

File: website/docs/reference/integrations/datadog.md [hunk 1]
```diff
@@ -16,7 +16,6 @@ The Datadog integration will perform a single retry if the HTTP POST against the
 You can choose to trigger updates for the following events:
 
 - feature-created
-- feature-updated (*)
```
Comment (nunogois) [inline website/docs/reference/integrations/datadog.md:19]: Maybe we should keep this one in the docs while it's still present in the `datadog-definition`?

File: website/docs/reference/integrations/datadog.md [hunk 2]
```diff
@@ -28,19 +27,40 @@ You can choose to trigger updates for the following events:
 - feature-stale-off
 - feature-environment-enabled
 - feature-environment-disabled
+- feature-variants-updated
```
Comment (nunogois) [inline website/docs/reference/integrations/datadog.md:30]: Possibly out of scope for this PR but I think we should use `feature-environment-variants-updated` instead: https://github.com/Unleash/unleash/pull/4801#discussion_r1332094112

File: website/docs/reference/integrations/datadog.md [hunk 3]
```diff
@@ -28,19 +27,40 @@ You can choose to trigger updates for the following events:
...
+- feature-potentially-stale-on
 
-> *) Deprecated, and will not be used after transition to environments in Unleash v4.3
-
-#### Parameters {#parameters}
+### Parameters {#parameters}
```
Comment (nunogois) [inline website/docs/reference/integrations/datadog.md:33]: I wonder about this change - Was it on purpose?

File: website/docs/reference/integrations/datadog.md [hunk 4]
```diff
@@ -28,19 +27,40 @@ You can choose to trigger updates for the following events:
...
-- **DD API KEY** - This is a required property.
+
+
+**Datadog API key** - This is a required property. The API key to use to authenticate with Datadog.
+
+**Datadog Source Type Name** - This property is optional. Sets source_type_name parameter to be included in Datadog events.
+
+**Extra HTTP Headers** This property is optional. Used to set the additional headers when unleash communicates with Datadog.
+
+**Body template** This property is optional. Used to override the body template used by Unleash when performing the HTTP POST. You may format you message using a [Mustache template](https://mustache.github.io). You will have the [Unleash event format](/reference/api/legacy/unleash/admin/events) available in the rendering context.
```
Comment (nunogois) [inline website/docs/reference/integrations/datadog.md:50]: Any reason why these items are no longer a list?

File: website/docs/reference/integrations/datadog.md [hunk 5]
```diff
@@ -28,19 +27,40 @@ You can choose to trigger updates for the following events:
...
   - EU: https://app.datadoghq.eu/api/v1/events
   - US1: https://app.datadoghq.com/api/v1/events
   - US3: https://us3.datadoghq.com/api/v1/events
   - US1-FED: https://app.ddog-gov.com/api/v1/events
-- **DD API KEY** - This is a required property.
+
+
+**Datadog API key** - This is a required property. The API key to use to authenticate with Datadog.
+
+**Datadog Source Type Name** - This property is optional. Sets source_type_name parameter to be included in Datadog events.
```
Comment (nunogois) [inline website/docs/reference/integrations/datadog.md:46]:
```suggestion
**Datadog Source Type Name** - This property is optional. Sets `source_type_name` parameter to be included in Datadog events.
```

File: website/docs/reference/integrations/datadog.md [hunk 6]
```diff
@@ -28,19 +27,40 @@ You can choose to trigger updates for the following events:
...
 
-> *) Deprecated, and will not be used after transition to environments in Unleash v4.3
-
-#### Parameters {#parameters}
+### Parameters {#parameters}
 
 Unleash Datadog integration takes the following parameters.
 
-- **Datadog Events URL** - This property is optional. The default url is https://api.datadoghq.com/api/v1/events. Needs to be changed if you are not not on the US1 [Datadog site](https://docs.datadoghq.com/getting_started/site/). Possible alternatives:
+**Datadog Events URL** - This property is optional. The default url is https://api.datadoghq.com/api/v1/events. Needs to be changed if you are not not on the US1 [Datadog site](https://docs.datadoghq.com/getting_started/site/). Possible alternatives:
```
Comment (nunogois) [inline website/docs/reference/integrations/datadog.md:37]:
```suggestion
**Datadog Events URL** - This property is optional. The default URL is https://api.datadoghq.com/api/v1/events. Needs to be changed if you are not not on the US1 [Datadog site](https://docs.datadoghq.com/getting_started/site/). Possible alternatives:
```

File: website/docs/reference/integrations/datadog.md (hunk 4 above)
Comment (nunogois) [inline website/docs/reference/integrations/datadog.md:50]:
```suggestion
**Body template** - This property is optional. Used to override the body template used by Unleash when performing the HTTP POST. You may format you message using a [Mustache template](https://mustache.github.io). You will have the [Unleash event format](/reference/api/legacy/unleash/admin/events) available in the rendering context.
```

File: website/docs/reference/integrations/datadog.md [hunk 7]
```diff
@@ -28,19 +27,40 @@ You can choose to trigger updates for the following events:
...
+```mustache
+{
+  "event": "{{event.type}}",
+  "createdBy": "{{event.createdBy}}",
+  "featureToggle": "{{event.data.name}}",
+  "timestamp": "{{event.data.createdAt}}"
+}
+```
+
+If you don't specify anything Unleash will send a formatted markdown body.
```
Comment (nunogois) [inline website/docs/reference/integrations/datadog.md:63]: Does it make sense to add an example here?

File: website/docs/reference/integrations/datadog.md (hunk 4 above)
Comment (daveleek) [reply website/docs/reference/integrations/datadog.md:50]: Thank you! omission on my part!

Comment (daveleek) [reply website/docs/reference/integrations/datadog.md]: It looked weird in the preview and wasn't used in the reference doc I used, but I should obviously have looked at all of them so I'll change it back

File: website/docs/reference/integrations/datadog.md (hunk 3 above)
Comment (daveleek) [reply website/docs/reference/integrations/datadog.md:33]: Yeah it was to differentiate the header from the emboldened parameter headers - they looked the same. Might be improved if I put the original bullet point lists back in

File: website/docs/reference/integrations/datadog.md (hunk 7 above)
Comment (daveleek) [reply website/docs/reference/integrations/datadog.md:63]: absolutely!

**Review by nunogois — APPROVED** [review]
LGTM, but I would feel more comfortable having @thomasheartman have a look before we merge this 👍

File: website/docs/reference/integrations/datadog.md (hunk 2 above)
Comment (thomasheartman) [reply website/docs/reference/integrations/datadog.md:31]: How does it work today? **Does** this work? If we should use `feature-environment-variants-updated`, can we switch this line to that?

```suggestion
- feature-environment-variants-updated
```

File: website/docs/reference/integrations/datadog.md [hunk 8]
```diff
@@ -28,19 +28,57 @@ You can choose to trigger updates for the following events:
...
+- feature-potentially-stale-on
 
 > *) Deprecated, and will not be used after transition to environments in Unleash v4.3
 
 #### Parameters {#parameters}
 
 Unleash Datadog integration takes the following parameters.
 
-- **Datadog Events URL** - This property is optional. The default url is https://api.datadoghq.com/api/v1/events. Needs to be changed if you are not not on the US1 [Datadog site](https://docs.datadoghq.com/getting_started/site/). Possible alternatives:
+- **Datadog Events URL** - This property is optional. The default URL is https://api.datadoghq.com/api/v1/events. Needs to be changed if you are not not on the US1 [Datadog site](https://docs.datadoghq.com/getting_started/site/). Possible alternatives:
```
Comment (thomasheartman) [inline website/docs/reference/integrations/datadog.md:40]: Are the only alternatives the ones listed here or could there be other ones too? Maybe something like this?

```suggestion
- **Datadog Events URL** - This property is optional. The default URL is https://api.datadoghq.com/api/v1/events. If you are not not using the US1 [Datadog site](https://docs.datadoghq.com/getting_started/site/), you'll need to change this. Some instances and their URLs are:
```

File: website/docs/reference/integrations/datadog.md [hunk 9]
```diff
@@ -28,19 +28,57 @@ You can choose to trigger updates for the following events:
...
+
+- **Datadog Source Type Name** - This property is optional. Sets `source_type_name` parameter to be included in Datadog events.
+
+- **Extra HTTP Headers** - This property is optional. Used to set the additional headers when Unleash communicates with Datadog.
+
+Example:
+
+```json
+{
+  "ISTIO_USER_KEY": "hunter2",
```
Comment (thomasheartman) [inline website/docs/reference/integrations/datadog.md:57]: Does it make sense to send an istio user key to datadog? Maybe we have a more relevant http header to show?

File: website/docs/reference/integrations/datadog.md [hunk 10]
```diff
@@ -28,19 +28,57 @@ You can choose to trigger updates for the following events:
...
+}
+```
+
+If you don't specify anything Unleash will send a formatted markdown body.
+
+Example:
+
+```markdown
+username created feature toggle (featurename)[http://your.url/projects/projectname/features/featurename] in project *projectname*
+```
```
Comment (thomasheartman) [inline website/docs/reference/integrations/datadog.md:81]: I think the availability not can go down here to call it out. We can also say that it's currently under development and expected to go out in the next few releases, but I think this is fine.

```suggestion
```

:::info Body template availability

The body template property will be introduced as a beta feature in Unleash 5.5 and is expected to be made generally available in Unleash 5.6.

:::
```

File: website/docs/reference/integrations/datadog.md [hunk 11]
```diff
@@ -28,19 +28,57 @@ You can choose to trigger updates for the following events:
...
+Example:
+
+```json
+{
+  "ISTIO_USER_KEY": "hunter2",
+  "SOME_OTHER_CUSTOM_HTTP_HEADER": "SOMEVALUE",
+}
+```
+
+- **Body template** - This property will be introduced as `beta` in `5.5` and is expected to enter `GA` in `5.6`. It is optional. Used to override the body template used by Unleash when performing the HTTP POST. You may format you message using a [Mustache template](https://mustache.github.io). You will have the [Unleash event format](/reference/api/legacy/unleash/admin/events) available in the rendering context.
```
Comment (thomasheartman) [inline website/docs/reference/integrations/datadog.md:62]: Suggestion: combination of moving availability and some slight rewording.

But what is the "unleash event format"? What are you linking to that page of the docs for? If you're looking for information on event shapes, they're on [`/reference/event-types`](https://docs.getunleash.io/reference/event-types) instead. I don't think I quite understand what you're saying. Is it that you'll have the event's properties available? In that case, maybe something like this?

```suggestion
- **Body template** - This is in optional property. The template is used to override the body template used by Unleash when performing the HTTP POST. You can format your message using a [Mustache template](https://mustache.github.io). Refer to the [Unleash event types](/reference/event-types) reference to find out which event properties you have access to in the template.
- ```

**Review by thomasheartman — APPROVED** [review]
Nice one 👏🏼  A couple minor suggestions, but I trust you to handle them 😄

File: website/docs/reference/integrations/datadog.md (hunk 2 above)
Comment (nunogois) [reply website/docs/reference/integrations/datadog.md]: I don't think it does anymore. My latest `feature-variants-updated` events from my local instance are from 2022 and they predate variants per environment. I suggest simply making the change, addressed in https://github.com/Unleash/unleash/pull/4802/commits/dbfeef1e23556408e9922e95f377f93a488c4f82

File: website/docs/reference/integrations/datadog.md (hunk 9 above)
Comment (nunogois) [reply website/docs/reference/integrations/datadog.md]: Fair. Made this more generic, both in the field placeholder and the example in the docs: https://github.com/Unleash/unleash/pull/4802/commits/afed94f69b2220d3e2101c2865441f9b5dbcf31f

File: website/docs/reference/integrations/datadog.md (hunk 10 above)
Comment (nunogois) [reply website/docs/reference/integrations/datadog.md:87]: I'm a bit confused by this comment but I think I understand the core concern of having the availability displayed as usual. Tried addressing it here: https://github.com/Unleash/unleash/pull/4802/commits/fdcc22a0c8bd170c75cc9277e4c772f9503d5745

File: website/docs/reference/integrations/datadog.md (hunk 11 above)
Comment (nunogois) [reply website/docs/reference/integrations/datadog.md]: Agreed. I think this was simply a copy-paste from `webhook.md`, so maybe we should update that one as well at some point. Addressed in https://github.com/Unleash/unleash/pull/4802/commits/a77a3698518082dc51032194bc28b89d4b60e5dc

**Comment by nunogois** [comment]
Thank you @daveleek and @thomasheartman - I'll merge this for now but if you notice anything wrong please let me know and I'll try to quickly fix it in a separate PR.

File: website/docs/reference/integrations/datadog.md (hunk 10 above)
Comment (thomasheartman) [reply website/docs/reference/integrations/datadog.md:87]: You got it! Looks great 😄

**Pull request merged**